client/src/pages/trainer/courses/index.tsx

```diff
@@ -1,150 +1,61 @@
-      'https://campustechnology.com/-/media/EDU/CampusTechnology/2019-Images/20190710onlinelearning.jpg',
-    title: 'Vue Mastery',
-    description: 'Lorem, ipsum dolor sit amet consectetur adipisicing elit. Cumque, aliquam.',
-    ratings: 5.0,
-    categories: [
-      { id: 1, name: 'JavaScript' },
-      { id: 2, name: 'TypeScript' },
-      { id: 4, name: 'React.js' },
-    ],
-    is_active: false,
-  },
-  {
-    id: 6,
-    image:
-      'https://res.cloudinary.com/highereducation/images/f_auto,q_auto/v1664380340/AccreditedSchoolsOnline.org/GettyImages-1221204650_8194983b6/GettyImages-1221204650_8194983b6.jpg ',
-    title: 'Vue Mastery',
-    description: 'Lorem, ipsum dolor sit amet consectetur adipisicing elit. Cumque, aliquam.',
-    ratings: 3.4,
-    categories: [
-      { id: 1, name: 'JavaScript' },
-      { id: 2, name: 'TypeScript' },
-      { id: 4, name: 'React.js' },
-    ],
-    is_active: false,
-  },
-  {
-    id: 7,
-    image: 'https://cdn.idropnews.com/wp-content/uploads/1970/01/23083056/MacBook-Pro.jpg',
-    title: 'Vue Mastery',
-    description: 'Lorem, ipsum dolor sit amet consectetur adipisicing elit. Cumque, aliquam.',
-    ratings: 2.2,
-    categories: [
-      { id: 1, name: 'JavaScript' },
-      { id: 2, name: 'TypeScript' },
-      { id: 4, name: 'React.js' },
-    ],
-    is_active: false,
-  },
-  {
-    id: 8,
-    image:
-      'https://www.shortform.com/blog/wp-content/uploads/2021/10/smarphone-apple-iphone-laptop-technology-750x350.jpg',
-    title: 'Vue Mastery',
-    description: 'Lorem, ipsum dolor sit amet consectetur adipisicing elit. Cumque, aliquam.',
-    ratings: 3.2,
-    categories: [
-      { id: 1, name: 'JavaScript' },
-      { id: 2, name: 'TypeScript' },
-      { id: 4, name: 'React.js' },
-    ],
-    is_active: false,
-  },
-  {
-    id: 9,
-    image:
-      'https://www.simplilearn.com/ice9/free_resources_article_thumb/Top_18_Technical_Courses_After_Graduation.jpg',
-    title: 'Vue Mastery',
-    description: 'Lorem, ipsum dolor sit amet consectetur adipisicing elit. Cumque, aliquam.',
-    ratings: 1.3,
-    categories: [
-      { id: 1, name: 'JavaScript' },
-      { id: 2, name: 'TypeScript' },
-      { id: 4, name: 'React.js' },
-    ],
-    is_active: false,
-  },
-];
+const CoursesListPage: React.FC = () => {
+  const router = useRouter();
+  const [page, setPage] = useState(1);
+  const [search, setSearch] = useState('');
 
-const handleSearch = (query: string): void => {
-  alert(`You Searched "${query}"`);
-};
+  const {
+    data: { results: courses = [], totalPages, current_page_number: currentPage } = {},
+    isLoading,
+    error,
+    refetch,
+  } = useGetCoursesQuery({ search, page });
 
-const CoursesListPage: React.FC = () => {
-  const [currentPage, setCurrentPage] = useState(1);
+  useEffect(() => {
+    if (search === '') {
+      const { query } = router;
+      delete query.search;
+      void router.replace({ query });
+    }
+  }, [search, refetch]);
```

Comment:
```suggestion
  }, [search]);
```
Please remove it from useEffect deps as well, and also is its not used in any other places, please remove it from line 21